packages/nuxt/src/app/nuxt.ts

```diff
@@ -331,26 +339,44 @@ export async function applyPlugin (nuxtApp: NuxtApp, plugin: Plugin & ObjectPlug
 }
 
 export async function applyPlugins (nuxtApp: NuxtApp, plugins: Array<Plugin & ObjectPlugin<any>>) {
+  const pluginMap: Record<string, any> = {}
   const parallels: Promise<any>[] = []
   const errors: Error[] = []
+
   for (const plugin of plugins) {
     if (import.meta.server && nuxtApp.ssrContext?.islandContext && plugin.env?.islands === false) { continue }
-    const promise = applyPlugin(nuxtApp, plugin)
+    const promise = plugin.dependsOn
+      ? Promise.all(plugin.dependsOn.map((name) => {
+        const pluginPromise = pluginMap[name]
+        if (import.meta.dev && !pluginPromise) {
+          console.warn(`[nuxt] ${plugin._name ? `Plugin "${plugin._name}"` : 'A plugin'} depends on "${name}" but "${name}" plugin does not exist or hasn't been registered yet.`)
```

Comment:
Should we avoid this possibility entirely? Perhaps we eagerly register a map with callbacks or similar? Or otherwise we are dependent on only depending on plugins alphabetically prior. (Or if not, we might advise here to use `order`.)